### PR TITLE
Add `eos_vrf_global` resource module

### PR DESCRIPTION
**Pull request by Ruchip16**
##### SUMMARY

This PR adds support for `eos_vrf_global` resource module

##### DESCRIPTION

 Here is the [model](https://github.com/ansible-network/resource_module_models/pull/277) design for the `eos_vrf_global` resource module.

##### ISSUE TYPE

- Feature Pull Request
- New Module Pull Request

##### COMPONENT NAME

`arista.eos.eos_vrf_global`

##### MILESTONES

- [x] Parsing entire argspec ( which covers gathered and parsed state )
- [x] Merged and Deleted state working
- [x] Replaced state working
- [x] Overridden state working
- [x]  Unit tests & Integration tests

### CODE REVIEW

File: plugins/modules/eos_vrf_global.py [hunk 1]
```diff
@@ -0,0 +1,461 @@
...
+    This output will always be in the same format as the
+    module argspec.
+after:
+  description: The resulting configuration after module execution.
+  returned: when changed
+  type: dict
+  sample: >
+    This output will always be in the same format as the
+    module argspec.
+commands:
```
Comment (rohitthakur2590) [inline plugins/modules/eos_vrf_global.py:395]: can we have actual example for this ?

File: plugins/modules/eos_vrf_global.py [hunk 2]
```diff
@@ -0,0 +1,461 @@
...
+# - description: this is sample vrf for feature testing
+#   name: my_vrf
+#   rd: '2:3'
+"""
+
+RETURN = """
+before:
+  description: The configuration prior to the module execution.
+  returned: when I(state) is C(merged), C(replaced), C(overridden), C(deleted) or C(purged)
+  type: dict
```
Comment (rohitthakur2590) [inline plugins/modules/eos_vrf_global.py:384]: as per the module documentation, this should be the list right

File: plugins/modules/eos_vrf_global.py [hunk 3]
```diff
@@ -0,0 +1,461 @@
...
+  sample: >
+    This output will always be in the same format as the
+    module argspec.
+after:
+  description: The resulting configuration after module execution.
+  returned: when changed
+  type: dict
```
Comment (rohitthakur2590) [inline plugins/modules/eos_vrf_global.py:391]: same as comment for `before`

File: tests/integration/targets/eos_vrf_global/tests/common/deleted.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
+---
+- ansible.builtin.debug:
+    msg: Start eos_vrf_global deleted integration tests connection={{ ansible_connection}}
+
+- ansible.builtin.include_tasks: _remove_config.yaml
+
+- ansible.builtin.include_tasks: _populate.yaml
+
+- block:
+    - name: Delete given vrf configuration
```
Comment (rohitthakur2590) [inline tests/integration/targets/eos_vrf_global/tests/common/deleted.yaml:10]: can we have test for where we delete specific vrf instance ? or that is not supported with `delete`  and achieved through `replaced` ?

File: tests/integration/targets/eos_vrf_global/tests/common/empty_config.yaml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  ignore_errors: true
+  arista.eos.eos_vrf_global:
+    running_config:
+    state: parsed
+
+- ansible.builtin.assert:
+    that:
+      - result.msg == 'value of running_config parameter must not be empty for state parsed'
+
+- name: Purged with empty configuration should give appropriate error message
```
Comment (rohitthakur2590) [inline tests/integration/targets/eos_vrf_global/tests/common/empty_config.yaml:60]: what are we trying to assert with `purged`? I believe purged can work with empty config

File: tests/integration/targets/eos_vrf_global/tests/common/parsed.yaml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+---
+- ansible.builtin.debug:
+    msg: START eos_vrf_global parsed integration tests on connection={{ ansible_connection }}
+
+- name: Parse externally provided VRF configuration
```
Comment (rohitthakur2590) [inline tests/integration/targets/eos_vrf_global/tests/common/parsed.yaml:5]: we need to assert the parsed output right ?

**Review by rohitthakur2590 — CHANGES_REQUESTED** [review]

File: tests/integration/targets/eos_vrf_global/tests/common/empty_config.yaml (hunk 5 above)
Comment (Ruchip16) [reply tests/integration/targets/eos_vrf_global/tests/common/empty_config.yaml:60]: true, removing that test for purged, we dont need it!

File: tests/integration/targets/eos_vrf_global/tests/common/parsed.yaml (hunk 6 above)
Comment (Ruchip16) [reply tests/integration/targets/eos_vrf_global/tests/common/parsed.yaml:5]: yep, added!

File: tests/integration/targets/eos_vrf_global/tests/common/deleted.yaml (hunk 4 above)
Comment (Ruchip16) [reply tests/integration/targets/eos_vrf_global/tests/common/deleted.yaml:10]: no we can do that using deleted as well!

File: plugins/modules/eos_vrf_global.py (hunk 1 above)
Comment (Ruchip16) [reply plugins/modules/eos_vrf_global.py:394]: i completely ignored this section, updating it!

File: plugins/modules/eos_vrf_global.py [hunk 7]
```diff
@@ -0,0 +1,461 @@
...
+short_description: Resource module to configure VRF definitions.
+description: This module provides declarative management of VRF definitions on Arista EOS platforms.
+version_added: 10.0.0
+author: Ruchi Pakhle (@Ruchip16)
+notes:
+- Tested against Arista EOS 4.23.0F
+- This module works with connection C(network_cli). See the L(EOS Platform Options,eos_platform_options).
+options:
+  config:
+    description: A dictionary containing device configurations for VRF, including a list of VRF definitions.
```
Comment (rohitthakur2590) [inline plugins/modules/eos_vrf_global.py:27]:
```suggestion
    description: A list of dictionaries containing device configurations for VRF definitions.
```

**Review by rohitthakur2590 — CHANGES_REQUESTED** [review]

**Review by rohitthakur2590 — APPROVED** [review]

**Pull request merged**